dex/bip-id.go

```diff
@@ -21,6 +26,31 @@ func BipSymbolID(symbol string) (uint32, bool) {
 	return idx, found
 }
 
+// BipTokenSymbolNetworks takes a asset part of a token's symbol
+// and returns a mapping of the various networks the token is
+// hosted on to their BIP-IDs.
+// For example if we have {"usdc.eth":60001, "usdc.matic":60002},
+// then passing in "usdc" will return {"eth": 60001, "matic": 60002}.
+func BipTokenSymbolNetworks(symbol string) (map[string]uint32, bool) {
+	if tokenSymbolNetworkBipIDs == nil {
+		tokenSymbolNetworkBipIDs = make(map[string]map[string]uint32)
+		for idx, sym := range bipIDs {
+			parts := strings.Split(sym, ".")
+			if len(parts) != 2 {
+				continue
+			}
+			tokenSym, network := parts[0], parts[1]
+			if _, found := tokenSymbolNetworkBipIDs[tokenSym]; !found {
+				tokenSymbolNetworkBipIDs[tokenSym] = make(map[string]uint32)
+			}
+			tokenSymbolNetworkBipIDs[tokenSym][network] = idx
+		}
+	}
+
+	networkIDs, found := tokenSymbolNetworkBipIDs[symbol]
+	return networkIDs, found
+}
```

Comment:
Since `bipIDs` is static, why not
```Go
var TokenChains = make(map[string][]uint32)

func init() {
	for _, symbol := range bipIDs {
		parts := strings.Split(symbol, ".")
		if len(parts) < 2 {
			continue
		}
		tokenSymbol, chainSymbol := parts[0], parts[1]
		chainID, found := BipSymbolID(chainSymbol)
		if !found {
			panic(fmt.Sprintf("malformed asset symbol %q", symbol))
		}
		TokenChains[tokenSymbol] = append(TokenChains[tokenSymbol], chainID)
	}
}
```
just once instead